hcitool: add page (#6311)
tldr-pages/tldr: pages/linux/hcitool.md

# hcitool

> > Monitor & Configure Bluetooth connections.
> > Send special commands to Bluetooth devices.

Reviewer: ```suggestion
> Monitor, configure connections and send special commands to Bluetooth devices.
> More information: <https://manned.org/hcitool>.
```

Author: Thanks for the suggestions @CleanMachine1

> - Scan for remote devices:

Reviewer 2: Just to make sure everyone knows it's about Bluetooth devices
```suggestion
- Scan for Bluetooth devices:
```

Author: Since this tool is only used by people who are in need
of this tool and as they definitely know that this tool is
used to work with Bluetooth, I haven't mentioned it clearly.
But definitely this is a good suggestion.

Thanks

`hcitool scan`

> - Output name of the device with its mac address:

Reviewer 2: ```suggestion
- Output the name of a device specifying its MAC address:
```

> - Output the name of a device specifying its MAC address:

Reviewer: ```suggestion
- Output the name of a device, returning its MAC address:
```

Reviewer: Wait does this give the mac address, or want the mac address?

Author: It give the mac address.

`hcitool name {{bdaddr}}`

> - Fetch info of a remote bluetooth device:

Reviewer 2: ```suggestion
- Fetch information about a remote Bluetooth device:
```

`hcitool info {{bdaddr}}`

> - Check link quality:

Reviewer 2: ```suggestion
- Check the link quality to a Bluetooth device:
```

`hcitool lq {{bdaddr}}`

> - Display transmit power level:

Reviewer 2: ```suggestion
- Display the transmit power level:
```

> `hcitool tpl {{bdaddr}} {{type}}`

Reviewer 2: Even after reading the man page, I don't really understand what `type` is.

Author: ```
       tpl <bdaddr> [type]
              Display transmit power level for the connection to the device with Bluetooth address bdaddr.

              The type can be 0 for the current transmit power level (which is default) or 1 for the maximum transmit power level.
```
Yeah, it's a bit confusing even in man-page.
What I've understood from this explanation is, if we specify 0, then the
tpl will be set to the current transmit power level instead of increasing
or decreasing it's intensity of transmission, whereas maximum tpl will
be applied when this type is set to 1.

Even though this is not a good explanation
on <type>, I hope it makes better sense...

Reviewer 2: If specifying `type` modifies it instead of displaying it we should remove it. Or change the description.

Author: How about changing the placeholder of type?
```console
hcitool tpl {{bdaddr}} {{0|1}}
```

Reviewer 2: But this still modifies the transmit power level instead of printing it. Either we change the description to `Modify the transmit power level` or we remove this.

Author: Ahh, now I get it.
Let's change the description.

Author: Changes committed!

> - Check link policy:

Reviewer 2: ```suggestion
- Display the link policy:
```

`hcitool lp`

> - Request authentication:

Reviewer 2: ```suggestion
- Request authentication with a specific device:
```

`hcitool auth {{bdaddr}}`

> - Display local devices:

Reviewer 2: What are local devices exactly? Are those the Bluetooth adapters?

Author: Local devices  are basically devices with Bluetooth card(s) for transmitting/receiving data.
